src/Hl7.Fhir.R5.Core/Specification/Source/ZipSource.cs

```diff
@@ -39,18 +38,33 @@ public static ZipSource CreateValidationSource()
             return CreateValidationSource(path);
         }
 
-        private DirectorySource DirectorySource => FileSource as DirectorySource;
-
         #region IConformanceSource
-        /// <inheritdoc/>
-        public IEnumerable<ConceptMap> FindConceptMaps(string sourceUri = null, string targetUri = null)
-            => DirectorySource.FindConceptMaps(sourceUri, targetUri);
+        /// <summary>Returns a reference to an <see cref="IConformanceSource"/> that exposes the contents of the ZIP archive.</summary>
+        public IConformanceSource Source => new DirectorySourceOnCommon(FileSource);
+
+        public IEnumerable<string> ListResourceUris(ResourceType? filter = null) => Source.ListResourceUris(filter);
+        public IEnumerable<ConceptMap> FindConceptMaps(string? sourceUri = null, string? targetUri = null) => Source.FindConceptMaps(sourceUri, targetUri);
+        public NamingSystem? FindNamingSystem(string uniqueId) => Source.FindNamingSystem(uniqueId);
+
+        private class DirectorySourceOnCommon : IConformanceSource
+        {
+            public DirectorySourceOnCommon(CommonDirectorySource source)
+            {
+                Source = source;
+            }
+
+            public CommonDirectorySource Source { get; }
 
-        /// <inheritdoc/>
-        public NamingSystem FindNamingSystem(string uniqueId) => DirectorySource.FindNamingSystem(uniqueId);
+            public CodeSystem? FindCodeSystemByValueSet(string valueSetUri) => Source.FindCodeSystemByValueSet(valueSetUri);
```

Comment:
These methods can be pushed to `CommonZipSource` as generics (like ` protected internal T? FindNamingSystem<T>(string uniqueId) where T : Resource`)

Reply:
done!